source/extensions/filters/network/common/redis/codec.h

```diff
@@ -27,6 +27,10 @@ class RespValue {
   RespValue() : type_(RespType::Null) {}
   ~RespValue() { cleanup(); }
 
+  RespValue(const RespValue& other);     // copy constructor
+  RespValue& operator=(RespValue other); // copy assignment
```

Comment:
nit: should take `const RespValue&` as arg.

General question: Where are these three methods actually required? It's not obvious from looking at the code.

Reply:
Right now the copy constructor is used in the onAskRedirection() pool callback for the various command request types. In particular, we create a new RespValue with "asking" prepended to the original contents. The copy assignment method came along as part of the rule of 3, and frankly, the move constructor came along as part of the "rule of 5". The move assignment method caused compiler ambiguous method resolution errors, so I got rid of it. I'll delete the unused methods, and add an explicit unit test for the copy constructor.

Reply:
The copy assignment method is also used by the std::vector::insert() call, so I'll keep it and fix its signature (thx for the catch).